### PR TITLE
Fix broken extension point to add replace home page

### PR DESCRIPTION
**Pull request by nwmac**
### Summary
Fixes #

### Checklist
- [ ] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [ ] The PR has a Milestone 
- [ ] The PR template has been filled out
- [ ] The PR has been self reviewed 
- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: shell/core/plugin.ts [hunk 1]
```diff
@@ -132,6 +132,11 @@ export class Plugin implements IPlugin {
         console.warn(`Layouts have been deprecated. We still have parent routes which use the same name and styling as the previous layouts. You should specify a parent, we're currently setting the parent to 'default'`); // eslint-disable-line no-console
         parentOverride = 'default';
       }
+
+      // Fix for Home page components with wrong layout - need to ensure the parentOverride is set
+      if (typelessRoute.component && typelessRoute.name === 'home' && typelessRoute.path === '/home') {
+        parentOverride = 'home_layout';
```
Comment (richard-cox) [inline shell/core/plugin.ts:138]: The warning above mentions we're setting the override to default, but then here we're silently setting to home_layout.

Should this be in an if/else above with a similar but more specific warning log?

**Review by richard-cox — CHANGES_REQUESTED** [review]
So we broke any page override that now uses a new template (where there was not like-for-like layout specified)? I had a quick look through routes.js, this might be the only one @codyrancher?

File: shell/core/plugin.ts (hunk 1 above)
Comment (nwmac) [reply shell/core/plugin.ts:138]: @richard-cox The home page component we use in our example uses the 'plain' layout - but this is NOT the parent for the home page route, so that is why the mechanism is broken.

The layout for the home page route does not have a name - possibly there is a work around using the empty parent route that I will check - but adding an explicit name of 'home_layout' makes sense.

I kept the change simple - because the home page is in the 'home_layout', the parent must be 'home_layout' in order to replace it.

For 2.10, we could take a more breaking change approach by adding the layout name 'home_layout' and then updating the 'setHomeComponent' or whatever it is to work correctly - that will break extensions that use the `addRoute` method though.

TL;DR - I personally think this is okay - @codyrancher can weigh in

**Comment by nwmac** [comment]
BTW. @richard-cox @codyrancher without adding the name to the top-level route that the home page uses, I couldn't see a workaround, since the name is `undefined` and trying to pass an undefined parent route name fails.

**Comment by codyrancher** [comment]
> So we broke any page override that now uses a new template (where there was not like-for-like layout specified)? I had a quick look through routes.js, this might be the only one @codyrancher?

Yes, we deprecated it over 7 months ago

> So we broke any page override that now uses a new template (where there was not like-for-like layout specified)? I had a quick look through routes.js, this might be the only one @codyrancher?

Correct

File: shell/core/plugin.ts (hunk 1 above)
Comment (codyrancher) [reply shell/core/plugin.ts:138]: I can't think of a less invasive fix so I'm fine with it.

**Review by codyrancher — APPROVED** [review]

File: shell/core/plugin.ts (hunk 1 above)
Comment (richard-cox) [reply shell/core/plugin.ts:138]: the approach is good! just wondered if we could avoid the misleading console message.

my suggestion was bad, adding the condition in the middle of the others work message wise but breaks functionality

**Review by richard-cox — APPROVED** [review]
